File: app/src/main/java/com/soft/credit911/ui/Chat/Fragment/Adapter/MessageDetailsAdapter.java
package com.soft.credit911.ui.Chat.Fragment.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.soft.credit911.ui.Chat.Fragment.Model.ChatObject;
import com.soft.credit911.R;

import java.util.ArrayList;

public class MessageDetailsAdapter extends RecyclerView.Adapter<MessageViewHolder> {
    public ArrayList<ChatObject> chatObjects = new ArrayList<>();
    Context context;

    public void add(ArrayList<ChatObject> chatObjects) {
        this.chatObjects = chatObjects;
        notifyDataSetChanged();
    }

    public MessageDetailsAdapter(Context context) {
        this.context = context;
    }

    @Override
    public MessageViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View itemView;
        switch (viewType) {
            case ChatObject.SEND_MESSAGE:
                itemView = inflater.inflate(R.layout.send_message, parent, false);
                return new SendMessageHolder(itemView);

            case ChatObject.RECEIVED_MESSAGE:
                itemView = inflater.inflate(R.layout.received_message, parent, false);
                return new ReceivedMessageHolder(itemView,context);

            default:
                itemView = inflater.inflate(R.layout.received_message, parent, false);
                return new ReceivedMessageHolder(itemView, context);
        }
    }

    @Override
    public void onBindViewHolder(MessageViewHolder holder, int position) {
        holder.onBindView(chatObjects.get(position));

    }

    @Override
    public int getItemViewType(int position) {
        return chatObjects.get(position).getType();
    }

    @Override
    public int getItemCount() {
        return (chatObjects == null) ? 0 : chatObjects.size();
    }
}

File: app/src/main/java/com/soft/credit911/adaptor/MessageDetailsAdapter.kt
package com.soft.credit911.adaptor

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import com.soft.credit911.datamodel.ChatObject
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.soft.credit911.R
import java.util.ArrayList

class MessageDetailsAdapter(var context: Context) : RecyclerView.Adapter<MessageViewHolder>() {
    var chatObjects: ArrayList<ChatObject>? = ArrayList()
    fun add(chatObjects: ArrayList<ChatObject>?) {
        this.chatObjects = chatObjects
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MessageViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val itemView: View
        return when (viewType) {
            ChatObject.SEND_MESSAGE -> {
                itemView = inflater.inflate(R.layout.send_message, parent, false)
                SendMessageHolder(itemView)
            }
            ChatObject.RECEIVED_MESSAGE -> {
                itemView = inflater.inflate(R.layout.received_message, parent, false)
                ReceivedMessageHolder(itemView, context)
            }
            else -> {
                itemView = inflater.inflate(R.layout.received_message, parent, false)
                ReceivedMessageHolder(itemView, context)
            }
        }
    }

    override fun onBindViewHolder(holder: MessageViewHolder, position: Int) {
        holder.onBindView(chatObjects!![position])
    }

    override fun getItemViewType(position: Int): Int {
        return chatObjects!![position].type
    }

    override fun getItemCount(): Int {
        return if (chatObjects == null) 0 else chatObjects!!.size
    }
}